plugins/ltac/tacinterp.ml

```diff
@@ -162,17 +162,26 @@ let catching_error call_trace fail (e, info) =
     fail located_exc
   end
 
-let catch_error call_trace f x =
+let locate_if_not_already ?loc (e, info) =
+  match Loc.get_loc info with
+  | None   -> (e, Option.cata (Loc.add_loc info) info loc)
+  | Some l -> (e, info)
+
+let catch_error ?loc call_trace f x =
   try f x
   with e when CErrors.noncritical e ->
     let e = Exninfo.capture e in
+    let e = locate_if_not_already ?loc e in
     catching_error call_trace Exninfo.iraise e
 
-let wrap_error tac k =
-  if is_traced () then Proofview.tclORELSE tac k else tac
+let wrap_error ?loc tac k =
+  if is_traced () then Proofview.tclORELSE tac k
+  else Proofview.tclORELSE tac (fun exn ->
```

Comment:
I think that you can use `Proofview.plus` instead, it should be more efficient and semantically equivalent.

Reply:
@ppedrot: Using `tclOR` changes the backtrace (it duplicated some calls, e.g. in test `output/Errors.v`). I'm afraid that I have to renounce to this change, at least at the current time. If you think it is worth further investigations, this can probably be done in another PR.